### DISCUSSION TITLE
Upgrading to webpack 5 causing dynamic import not working

### DISCUSSION THREAD

**Discussion by oddguan** [webpack/webpack#16411] [Q&A]
Dear webpack community,

I was trying to upgrade my project from webpack 4 to 5. I followed the guide on the webpack official website, and so far the compilation works without errors. 
However I'm currently receiving a runtime error where it complains:
```
Uncaught ReferenceError: jsonpScriptSrc is not defined
```
I digged through the source-map and found out it was when I'm using dynamic import and jsonpScriptSrc was used, but it is not defined from anywhere.
I tried to google for an answer but couldn't find one. Just want to know if anyone has met the same issue before? Any help is greatly appreciated. Thank you!

**Comment by alexander-akait** [comment]
How you use `jsonpScriptSrc`? DO you have mixed code with webpack v4 and webpack v5?

**Reply by oddguan** [reply]
> How you use `jsonpScriptSrc`? DO you have mixed code with webpack v4 and webpack v5?

I am not using `jsonpScriptSrc` in any form. Tbh I don't even know what this function does. It just shows in my console with an error saying it is not defined in the outer scope. 
I assume the problem comes from I mixed webpack 4 and 5, but I'm not sure where I mixed it. Any suggestions on what should I look for?

**Reply by oddguan** [reply]
Some extra context:
The runtime error comes from here:
<img width="518" alt="image" src="https://user-images.githubusercontent.com/14129720/197679730-0cd9122a-6a9f-4ac0-9046-1ef1abc0e7d1.png">
Then I did a global search for `jsonpScriptSrc` on all the source files, and didn't find any definition of the function. I assume the problem comes from I didn't load a prerequisite chunk or something?

**Reply by alexander-akait** [reply]
@oddguan Can you create a minkum reproducible test repo, do you have any non offical plugins, it is hard to say without information where is this problem

**Reply by 2snEM6** [reply]
This is happening to me as well

**Reply by cipollafrancesco** [reply]
Hey,

**I ran into the same issue while upgrading an old repo to Webpack 5.**

At runtime, I was getting the following error:

```
ReferenceError: jsonpScriptSrc is not defined
```

After investigating, I discovered that the problem was caused by a dependency conflict in my `package.json`. Specifically, I had the `webpack-retry-chunk-load-plugin`, which had a peerDependency on Webpack 4.

To resolve it, I removed the plugin from my dependencies and from my production config (`webpack.prod.js`).

Here’s how it looked before the fix:

```js
new RetryChunkLoadPlugin({
  maxRetries: 3,
  retryDelay: 3000,
  // Optional: Add more specific configurations for RetryChunkLoadPlugin if needed
}),
```

After this change, the `jsonpScriptSrc` error disappeared. 
**I also confirmed by searching through the entire build folder, and the broken `jsonpScriptSrc` reference no longer appears.**

Hope this helps!
